paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -572,7 +572,11 @@ internal class DefaultFlowController @Inject internal constructor(
         paymentOptionResult: PaymentOptionResult?
     ) {
         paymentOptionResult?.paymentMethods?.let {
-            viewModel.state = viewModel.state?.copy(customerPaymentMethods = it)
+            val currentState = viewModel.state
```

Comment:
I know this isn't a behavior change of the PR, but I'm failing to see how this does anything in practice 🤔 

What does this do?

Reply:
Since the `state` attribute of `FlowControllerViewModel` is a mutable & nullable property, the Kotlin compiler cannot smart cast to a non-nullable when re-used in the same context to copy a non-nullable internal property because `state` itself is a mutable property.
```kotlin
viewModel.state = viewModel.state?.copy(
    // Smart cast to 'PaymentSheetState.Full' is impossible, because 'viewModel.state' is a mutable property that 
    // could have been changed by this time
    customer = viewModel.state?.copy(paymentMethods = it)
)
```

In general, I've also done this with non-mutable properties as well to hold the internal reference.

Reply:
Ohh, sorry. I'm not seeing how the `customer` object could change as a result of this call.

Wouldn't it always be the same before and after?

Reply:
Ah my bad! Looked at the changes related to it. This was done to account for changes made when in `PaymentSheet` after potentially removing or modifying cards.

Reply:
Ahh, removing! Ok, that makes sense! Thanks!